src/transformers/models/voxtral/configuration_voxtral.py

```diff
@@ -0,0 +1,254 @@
+        self.initializer_range = initializer_range
+        self.scale_embedding = scale_embedding  # scale factor will be sqrt(d_model) if True
+        self.max_source_positions = max_source_positions
+
+
+class VoxtralConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`VoxtralForConditionalGeneration`]. It is used to instantiate an
+    Qwen2-Audio model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the Qwen2-Audio.
+
+    e.g. [Qwen/Qwen2-Audio-7B](https://huggingface.co/Qwen/Qwen2-Audio-7B)
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        audio_config (`Union[AutoConfig, dict]`,  *optional*, defaults to `CLIPVisionConfig`):
+            The config object or dictionary of the audio backbone.
+        text_config (`Union[AutoConfig, dict]`, *optional*, defaults to `LlamaConfig`):
+            The config object or dictionary of the text backbone.
+        audio_token_index (`int`, *optional*, defaults to 151646):
+            The image token index to encode the image prompt.
+
+    Example:
+
+    ```python
+    >>> from transformers import VoxtralForConditionalGeneration, VoxtralConfig, VoxtralEncoderConfig, Qwen2Config
+
+    >>> # Initializing a VoxtralEncoder config
+    >>> audio_config = VoxtralEncoderConfig()
+
+    >>> # Initializing a Qwen2 config
+    >>> text_config = Qwen2Config()
+
+    >>> # Initializing a Voxtral configuration
+    >>> configuration = VoxtralConfig(audio_config, text_config)
+
+    >>> # Initializing a model from the qwen2-audio style configuration
+    >>> model = VoxtralForConditionalGeneration(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "voxtral"
+    sub_configs = {"text_config": AutoConfig, "audio_config": AutoConfig}
+
+    _default_text_config_kwargs = {
+        "vocab_size": 131072,
+        "max_position_embeddings": 131072,
+        "hidden_size": 3072,
+        "intermediate_size": 8192,
+        "num_hidden_layers": 30,
+        "num_attention_heads": 32,
+        "num_key_value_heads": 8,
+        "hidden_act": "silu",
+        "initializer_range": 0.02,
+        "rms_norm_eps": 1e-05,
+        "pretraining_tp": 1,
+        "use_cache": True,
+        "rope_theta": 100000000.0,
+        "rope_scaling": None,
+        "attention_bias": False,
+        "attention_dropout": 0.0,
+        "mlp_bias": False,
+        "head_dim": 128,
+        "tie_word_embeddings": False,
+        "bos_token_id": 1,
+        "eos_token_id": 2,
+        "sliding_window": None,
+    }
+
+    def __init__(
+        self,
+        audio_config=None,
+        text_config=None,
+        audio_token_id=24,
```

Comment:
Is it possible to force the user to provide this via the `config.json`. Bit worried here that future iterations of voxtral will have different audio token ids and we silently still set it to 24

Reply:
Yes for sure